File: IA/ASD/Lab 7/algo.py
def ShellSort(l):
    gap = len(l)//2
    while gap > 0:
        for i in range(gap,len(l)):
            temp = l[i]
            j = i
            while j >= gap and l[j-gap] > temp:
                l[j] = l[j-gap]
                j = j - gap
            l[j] = temp
        gap = gap//2
    return l

l=[10,1,2,7,4,8,9,3,5,6]

print("Lista originala: ", l)
print("Lista sortata:", ShellSort(l))

File: IA/ASD/Lab 7/algo.cpp
#include <bits/stdc++.h>
using namespace std;


//Vectorul cand e luat ca parametru e luat deja ca pointer si putem schimba valorile din el
void shellSort(int v[], int n) {
    int gap = n / 2;
    while (gap > 0) {
        for (int i = gap; i < n; i++) {
            int j = i;
            int temp = v[i];
            while (j >= gap && v[j - gap] > temp) {
                v[j] = v[j - gap];
                j -= gap;
            }
            v[j] = temp;
        }
        gap /= 2;
    }
}

int main() {
    int v[] = {10, 4, 8, 3, 5, 9, 7, 2, 6, 1};
    shellSort(v, 10);
    for (int i = 0; i < 10; i++) {
        cout << v[i] << " ";
    }
    return 0;
}
